### ISSUE TITLE
appendconfig does not work

### ISSUE THREAD

**Issue by JimmySnails** [libretro/RetroArch#5457]
It seems, that if a video_shader line is found in retroarch.cfg, the appended config is ignored.

the --appendconfig CLI argument worked fine before one of the last updates. on 1.6.3 it doesn't override retroarch.cfg anymore.

When i set up a core like genplusgx and i want to use one of the handheld shaders for GameGear and a CRT shader for Genesis, the last shader that has been written to retroarch.cfg is used, instead of the appended one. 

Steps to reproduce:

- Delete the video_shader="..." line from retroarch.cfg.
- Create a custom config file for gamegear / genesis with:
video_shader = ":\shaders\shaders_glsl\crt\crt-royale.glslp" 
and
video_shader = ":\shaders\shaders_glsl\handheld\console-border\gg-4x.glslp"
- Start gamegear game (border is displayed correctly)
retroarch.exe -L cores/genesis_plus_gx_libretro.dll --appendconfig "config\additional_config\gamegear.cfg" "d:\Emulatoren\roms\Addams Family, The (World).zip" --verbose
- Start Genesis Core (correct shader is not loaded)

For me that's a major drawback... i really like to use one core for multiple platforms with different shaders / save dirs / controller configs and so on...

**Comment by fr500** [comment]
Do you have anything in shaders/presets?

**Comment by JimmySnails** [comment]
I'm not sure what you mean? 
After upgrading my previous retroarch installation messed up the shader configs, i tried again with a clean setup, but still got the same issue. There are 12 files in shaders\shaders_glsl\presets
But i didn't put anything there.

**Comment by fr500** [comment]
Well, try overriding something else, like, display fps on and display fps off in the appendconfig or something.
You're saying it ignores the config, lets find out if it truly does or if it just ignores the shader changes.

**Comment by JimmySnails** [comment]
Thx for the clarification. It seems, that every additional config entry is always ignored, except if the line is missing in both retroarch.cfg and core.cfg.

Here are my test cases, i hope they are understandable...
In retroarch.cfg there's this line per default:
fps_show = "false"

If i add fps_show = "true" to gamegear.cfg (the --append config), no fps counter is shown.
if i add fps_show = "true" to Genesis Plus GX.cfg, the fps counter is shown.
if i add fps_show = "true" to Genesis Plus GX.cfg and fps_show=false to gamegear.cfg, the fps counter is shown.

If i remove the fps_show line from retroarch.cfg, set the  Genesis Plus GX.cfg to false and gamegear.cfg to true, no fps counter is shown. (but the line fps_show=false is written to retroarch.cfg again)

If i remove the fps_show line from retroarch.cfg and Genesis Plus GX.cfg and set it to true in gamegear.cfg, the FPS counter is shown and added to retroarch.cfg

if i add false back to Genesis Plus GX.cfg, (leave true in retroarch.cfg and gamegear.cfg) no FPS counter is shown.

**Comment by fr500** [comment]
I guess the append config switch is broken then?
I personally don't use it because overrides can do everything I need (and from the gui)

I may check this on the weekend but I just tried setting fps to on and saving an override for a game and it's working as expected.

**Comment by JimmySnails** [comment]
Maybe it's more readable if i put in this way:

retroarch.cfg | Genesis Plus GX.cfg | gamegear.cfg | FPS Counter shown:
--------------|-----------------------|-----------------|-----------------------
false              |     n/a                        |   true              |   NO
false|true|false| YES
n/a|false|true| NO
n/a|n/a|true| YES
true|false|true| NO
true|n/a|false| NO

**Comment by JimmySnails** [comment]
Yeah it seems it's broken... but i don't think it's possible to configure different overlays for different systems on the same core in another way? Per game configuration in a huge library isn't really a good approach... And i use this switch for GB / GBC and all the GenPlus GX systems. 

If you create a bountysource ticket, i'd gladly buy you a beer  for the fix! :)

**Comment by BubsFr** [comment]
Same behavior explained here: https://github.com/libretro/RetroArch/issues/5405

**Comment by fr500** [comment]
it's not the same at all

**Comment by BubsFr** [comment]
I think it's the same, I wrote it.
JimmySnails probably is better explaining what I was trying to explain...

**Comment by fr500** [comment]
@BubsFr no

@JimmySnails overrides and appendconfig are not meant to be mixed because it works like this:

Startup:
- RA loads main config
- RA appends config (THAT'S ALL of the append lifecycle)

On content load:
- RA loads content, looks for an override, if one exists it applies the settings from that config on top of the active config (no other files are read).

Thus overrides load LAST and appendconfig can't be enforced on top of them.
Appendconfig is a legacy option and it's only meant for CLI (or external frontend users) which are not our main focus.

**Comment by BubsFr** [comment]
@fr500 , On startup, is "RA appends config" supposed to override "main config" if values are defined in both? If yes, that do not work (at least for the ones I tried)

**Comment by JimmySnails** [comment]
@fr500 Now i'm really confused.... It seems then, appendconfig was faulty all before? It was loaded last and hence overrode the values one set there. 

But that is exactly what i need to use to retroarch properly. :(
There's no "override config" option for CLI, right? 

The problem is, that once anything is set in the override config, it's written to retroarch.cfg (even if one uses a core specific config that doesn't have those specific configs set, that are used in the appendconfig) and therefore not loaded anymore. 

Is appendconfig supposed, to override values from retroarch.cfg if a core specific config is used? 

If it's not able to override retroarch.cfg, it doesn't have any use, because first time any appendconfiig is loaded it's values are written to retroarch.cfg? So it cannot be used a second time, right?

@BubsFr I think you want to do exactly the same as me?

**Comment by dankcushions** [comment]
> If it's not able to override retroarch.cfg, it doesn't have any use, because first time any appendconfiig is loaded it's values are written to retroarch.cfg? So it cannot be used a second time, right?

you should have not have `config_save_on_exit = true` in your main config or override, if you don't want those settings to migrate.

**Comment by JimmySnails** [comment]
Thank you, that's a workaround i can live with!

**Comment by fr500** [comment]
Actually overrides disable config saving already while active.

Appendconfig doesn't though.

**Comment by keithbowes** [comment]
> Appendconfig is a legacy option 

Then why does `retroarch --help` indicate that -s and -S are deprecated in favor of using --appendconfig if --appendconfig is a legacy option and won't even work if there are other overrides?
